README
Gtk2-Notify

INSTALLATION

To install this module, run the following commands:

    perl Makefile.PL
    make
    make test
    make install


SUPPORT AND DOCUMENTATION

After installing, you can find documentation for this module with the perldoc command.

    perldoc Gtk2::Notify

You can also look for information at:

    Search CPAN
        http://search.cpan.org/dist/Gtk2-Notify

    CPAN Request Tracker:
        http://rt.cpan.org/NoAuth/Bugs.html?Dist=Gtk2-Notify

    AnnoCPAN, annotated CPAN documentation:
        http://annocpan.org/dist/Gtk2-Notify

    CPAN Ratings:
        http://cpanratings.perl.org/d/Gtk2-Notify

COPYRIGHT AND LICENCE

Copyright (C) 2006 Florian Ragwitz

This program is free software; you can redistribute it and/or modify it
under the same terms as Perl itself.


Changes
Revision history for Gtk2-Notify

0.04    Thu, 04 Oct 2007 14:26:02 +0200
        * Let set_hint() stringify it's argument when it doesn't know
          what to do with it.
        * Don't compile things with -Werror.
        * Be more portable by using ppport.h.
        * Several minor, not user-visible improvements.

0.03    Thu, 31 May 2007 13:48:03 +0200
        * Bind status icon functions.
        * Various improvements.

0.02    Mon, 28 Aug 2006 13:07:56 +0200
        * Make the last three arguments to ->new optional.
        * Make the last two arguments to ->update optional.
        * Make the userdata in add_action optional.
        * Also initialize Gtk2 when called with -init.
        * Port the pynotify examples.

0.01    Sun, 27 Aug 2006 17:52:24 +0200
        * First version, released on an unsuspecting world.


lib/Notify.pm
package Gtk2::Notify;

use strict;
use warnings;
use Gtk2;
require DynaLoader;

our @ISA = qw( DynaLoader );
our $VERSION = '0.04';

sub import {
    my $class = shift;
    my $init = 0;

    while (my $arg = shift @_) {
        if ($arg =~ /^-init$/) {
            my $app_name;
            if (!defined ($app_name = shift @_)) {
                require Carp;
                Carp::croak('-init requires the application name to use as its first argument');
            }

            Gtk2->init;
            Gtk2::Notify->init($app_name);
        } else {
            $class->VERSION($arg);
        }
    }
}

sub dl_load_flags { 0x01 }

__PACKAGE__->bootstrap($VERSION);

1;

__END__
=head1 NAME

Gtk2::Notify - Perl interface to libnotify

=head1 VERSION

Version 0.04

=head1 SYNOPSIS

    use Gtk2::Notify -init, "app_name";

    my $notification = Gtk2::Notify->new($summary, $message, $icon, $attach_widget);
    $notification->show;

=head1 INITIALISATION

    use Gtk2::Notify qw/-init app_name/;

=over

=item -init

Importing Gtk2::Notify with the -init option requires one additional argument:
the application name to use. This is equivalent to
Gtk2::Notify->init($app_name).

=back

=head1 AUTHOR

Florian Ragwitz, C<< <rafl at debian.org> >>

=head1 BUGS

Please report any bugs or feature requests to
C<bug-gtk2-notify at rt.cpan.org>, or through the web interface at
L<http://rt.cpan.org/NoAuth/ReportBug.html?Queue=Gtk2-Notify>.
I will be notified, and then you'll automatically be notified of progress on
your bug as I make changes.

=head1 SUPPORT

You can find documentation for this module with the perldoc command.

    perldoc Gtk2::Notify

You can also look for information at:

=over 4

=item * AnnoCPAN: Annotated CPAN documentation

L<http://annocpan.org/dist/Gtk2-Notify>

=item * CPAN Ratings

L<http://cpanratings.perl.org/d/Gtk2-Notify>

=item * RT: CPAN's request tracker

L<http://rt.cpan.org/NoAuth/Bugs.html?Dist=Gtk2-Notify>

=item * Search CPAN

L<http://search.cpan.org/dist/Gtk2-Notify>

=back

=head1 COPYRIGHT & LICENSE

Copyright 2006-2007 Florian Ragwitz, all rights reserved.

This library is free software; you can redistribute it and/or
modify it under the terms of the GNU Library General Public
License as published by the Free Software Foundation; either
version 2 of the License, or (at your option) any later version.

This library is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
Library General Public License for more details.

You should have received a copy of the GNU Library General Public
License along with this library; if not, write to the
Free Software Foundation, Inc., 59 Temple Place - Suite 330,
Boston, MA  02111-1307  USA.

=cut

1; # End of Gtk2::Notify


Makefile.PL
use strict;
use warnings;
use File::Spec;
use Glib::CodeGen;
use Glib::MakeHelper;
use ExtUtils::Depends;
use ExtUtils::PkgConfig;
use ExtUtils::MakeMaker;

mkdir 'build', 0777;

my %pkg_config = ExtUtils::PkgConfig->find('libnotify');

Glib::CodeGen->parse_maps('notify');
Glib::CodeGen->write_boot(ignore => qr/^Gtk2::Notify$/);

our $notify = ExtUtils::Depends->new('Gtk2::Notify', 'Gtk2');
our @xs_files = <xs/*.xs>;

$notify->add_xs( @xs_files );
$notify->add_typemaps( File::Spec->rel2abs( File::Spec->catfile(qw( build notify.typemap )) ) );
$notify->set_inc( $pkg_config{cflags}.' -Ibuild -Wall' );
$notify->set_libs( $pkg_config{libs} );
$notify->install( File::Spec->catfile(qw( build notify-autogen.h )) );
$notify->add_pm(
    'lib/Notify.pm' => '$(INST_LIBDIR)/Notify.pm',
);

WriteMakefile(
    NAME                => 'Gtk2::Notify',
    AUTHOR              => 'Florian Ragwitz <rafl@debian.org>',
    VERSION_FROM        => 'lib/Notify.pm',
    ABSTRACT_FROM       => 'lib/Notify.pm',
    PL_FILES            => {},
    PREREQ_PM => {
        'Test::More' => '0',
        'Glib'       => '1.093',
    },
    MAN3PODS => {
        'lib/Notify.pm' => '$(INST_MAN3DIR)/Gtk2::Notify.$(MAN3EXT)',
        Glib::MakeHelper->do_pod_files(@xs_files),
    },
    dist                => { COMPRESS => 'gzip -9f', SUFFIX => 'gz', },
    clean               => { FILES => 'Gtk2-Notify-*' },
    $notify->get_makefile_vars,
);

package MY;

sub postamble {
    return 'POD_SET=$$Glib::GenPod::COPYRIGHT="Copyright (C) 2006-2007 Florian Ragwitz<br><br>'
        . 'This library is free software; you can redistribute it and/or modify it under '
        . 'the terms of the GNU Library General Public License as published by the Free '
        . 'Software Foundation; either version 2 of the License, or (at your option) any '
        . 'later version.<br><br>'
        . 'This library is distributed in the hope that it will be useful, but WITHOUT ANY '
        . 'WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A '
        . 'PARTICULAR PURPOSE.  See the GNU Library General Public License for more details.'
        . '<br><br>'
        . 'You should have received a copy of the GNU Library General Public License along '
        . 'with this library; if not, write to the Free Software Foundation, Inc., 59 '
        . 'Temple Place - Suite 330, Boston, MA  02111-1307  USA.";'
        . Glib::MakeHelper->postamble_clean()
        . Glib::MakeHelper->postamble_docs_full(DEPENDS => $main::notify);
}
